perceval/utils/parameter.py

```diff
@@ -135,6 +138,7 @@ def set_value(self, v: float, force: bool = False):
         if self.fixed and not force:
             raise RuntimeError("cannot set fixed parameter", v, self._value)
         self._value = v
+        self._notify_expressions()
```

Comment:
That's a cool "observer" pattern you've implemented there. However, I'm wondering if that's not overkill for our needs. On top of that, as it's not totally "abstracted", it looks like a circular dependency (Parameter needs to know about Expression, and Expression needs to know about Parameter).

Alternatively, we could ask `Expression.__float__` to re-evaluate at every call. This would be a problem only if you're unsing the same expression in hundreds of different places at once.

In your wildest use case, in how many places are you using a single expression?